### PR TITLE
koord-scheduler: improve reservation compatibility

### PR DESCRIPTION
**Pull request by saintube**
Signed-off-by: saintube <saintube@foxmail.com>

### Ⅰ. Describe what this PR does

1. improve the compatibility of the resource reservation by
  - move reservation utility functions into `pkg/util`
  - add some helper functions for scheduler plugins easily supporting reservations

### Ⅱ. Does this pull request fix one issue?

#96 #188 

### V. Checklist

- [X] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [X] All checks passed in `make test`

### CODE REVIEW

File: pkg/scheduler/plugins/nodenumaresource/plugin.go [hunk 1]
```diff
@@ -373,26 +369,13 @@ func (p *Plugin) PreBind(ctx context.Context, cycleState *framework.CycleState,
...
-				klog.Error("Failed to patch Pod %s/%s, patch: %v, err: %v", pod.Namespace, pod.Name, string(patchBytes), err)
-			}
-			return err
-		})
+	// patch pod or reservation (if the pod is a reserve pod) with new annotations
+	patchData := &util.PatchData{}
+	patchData.SetAnnotations(pod.Annotations)
+	err = util.PatchPodOrReservation(p.handle, podOriginal, patchData)
 	if err != nil {
+		klog.V(4).InfoS("Failed to preBind Pod with CPUSet",
```
Comment (eahydra) [inline pkg/scheduler/plugins/nodenumaresource/plugin.go:377]: use ErrrorS?

Comment (saintube) [reply pkg/scheduler/plugins/nodenumaresource/plugin.go]: fixed

**Pull request merged**